ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -358,21 +377,21 @@ static void createCommandTopicIfNecessary(final KsqlRestConfig restConfig,
     }
   }
 
-  private static void loadUiWar() {
-    final File uiFolder = new File(UI_FOLDER);
+  private void loadUiWar() {
+    final File uiFolder = new File(this.uiFolder);
     log.info("Loading UI-WAR from {}", uiFolder.getAbsolutePath());
 
     final File[] files = uiFolder
         .listFiles((dir, name) -> name.endsWith(".war"));
 
     if (files != null) {
-      Arrays.stream(files).forEach(KsqlRestApplication::unzipWar);
+      Arrays.stream(files).forEach(war -> KsqlRestApplication.unzipWar(war, uiFolder));
     }
   }
 
-  private static void unzipWar(final File warFile) {
+  private static void unzipWar(final File warFile, File uiFolder) {
```

Comment:
uiFolder is a File - before it was a String, so the concat relies on the File.toString()....